File: GurmukhiFlutter/lib/models/source_type.dart
enum SourceType {
  jaapSahib('jaap_sahib', 'Jaap Sahib'),
  rehrasSahib('rehras_sahib', 'Rehras Sahib'),
  chaupaiSahib('chaupai_sahib', 'Chaupai Sahib'),
  sukhmaniSahib('sukhmani_sahib', 'Sukhmani Sahib'),
  hukamnama('hukamnama', 'Daily Hukamnama');

  const SourceType(this.value, this.displayName);
  final String value;
  final String displayName;

  static SourceType fromValue(String value) {
    return SourceType.values.firstWhere(
      (item) => item.value == value,
      orElse: () => SourceType.hukamnama,
    );
  }
}

File: Models/SourceType.swift
import Foundation

enum SourceType: String, Codable, CaseIterable, Identifiable {
    case jaapSahib = "jaap_sahib"
    case rehrasSahib = "rehras_sahib"
    case chaupaiSahib = "chaupai_sahib"
    case sukhmaniSahib = "sukhmani_sahib"
    case hukamnama = "hukamnama"

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .jaapSahib: return "Jaap Sahib"
        case .rehrasSahib: return "Rehras Sahib"
        case .chaupaiSahib: return "Chaupai Sahib"
        case .sukhmaniSahib: return "Sukhmani Sahib"
        case .hukamnama: return "Daily Hukamnama"
        }
    }
}
